tests/outlineCompiler_test.py

```diff
@@ -596,6 +610,65 @@ def test_optimized_default_but_no_nominal_widths(self, FontClass):
         assert private.defaultWidthX == 500
         assert private.nominalWidthX == 0
 
+    def test_underline_without_public_key(self, testufo):
+        # Test with no lib key
+        compiler = OutlineOTFCompiler(testufo)
+        compiler.compile()
+
+        post = compiler.otf["post"].underlinePosition
+
+        cff = compiler.otf["CFF "].cff
+        cff_underline = cff[list(cff.keys())[0]].UnderlinePosition
+
+        assert post == -200
+        assert cff_underline == -200
+
+    def test_underline_with_public_key(self, testufo):
+        # Test with a lib key and postscriptUnderlinePosition
```

Comment:
it is unclear that postscriptUnderlinePosition is already set on the testufo, apart from the hint in the comment above; maybe set it explicitly to something else so that you can assert it's as expected and it's different from the value of the lib key.

Also you may want to test explicitly the case where neither the lib key nor the info attribute is set, for completeness sake.

Reply:
Got it. Resolved.